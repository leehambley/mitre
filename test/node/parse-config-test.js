const path = require("path");
const fs = require("fs");
const mitre = require("../../src/ffi/node");

describe("parseConfig", () => {
  it.todo("throws errors on invalid configs");
  it("returns not-null for a not-null config file", () => {
    let p = path.resolve(
      __dirname,
      "../fixtures/example-1-simple-mixed-migrations/mitre.yml"
    );
    fs.accessSync(p, fs.F_OK); // assert we have the right file
    expect(mitre.parseConfig(p)).toBeTruthy()
  });
  
  it("returns a parsed config by dir", () => {
    let p = path.resolve(
      __dirname,
      "../fixtures/example-1-simple-mixed-migrations/mitre.yml"
    );
    let config = mitre.parseConfig(p)
    expect(
      config.migrationsDirectory
    ).toEqual(expect.stringMatching(/.*example-1-simple-mixed-migrations\/migrations\/$/));
    expect(config.configuredRunners).toMatchSnapshot();
  });
});