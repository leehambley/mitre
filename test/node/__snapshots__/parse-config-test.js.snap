// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseConfig returns a parsed config by dir 1`] = `
Object {
  "es-docker": Object {
    "_runner": "http",
    "database": "",
    "databaseNumber": 0,
    "index": "",
    "ipOrHostname": "es",
    "password": "",
    "port": 0,
    "username": "",
  },
  "es-mysql": Object {
    "_runner": "mysql",
    "database": "mitre",
    "databaseNumber": 0,
    "index": "",
    "ipOrHostname": "127.0.0.1",
    "password": "example",
    "port": 3306,
    "username": "root",
  },
  "es-postgres": Object {
    "_runner": "postgres",
    "database": "",
    "databaseNumber": 0,
    "index": "",
    "ipOrHostname": "postgres",
    "password": "example",
    "port": 5432,
    "username": "root",
  },
  "mitre": Object {
    "_runner": "mysql",
    "database": "mitre",
    "databaseNumber": 0,
    "index": "",
    "ipOrHostname": "127.0.0.1",
    "password": "example",
    "port": 3306,
    "username": "root",
  },
}
`;
